### DISCUSSION TITLE
Custom Credential Provider - Can't Build Windows 10.0.22000.0 CredentialProvider.Interop.dll

### DISCUSSION THREAD

**Discussion by DarrylIdle** [phaetto/windows-credentials-provider#17] [Q&A]
Hi,

When trying to follow the steps outlined to re-create the CredentialProvider.Interop.dll I can't seem to get it to properly build the dll. Using **Developer Command Prompt for VS 2019** for the MIDL portion works but when I go to run the tlbimp2.exe command it errors stating:

`TlbImp : error TI1000 : The type library importer encountered an unexpected exception: System.BadImageFormatException - An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)`

Commands being used:
`midl /x64 /target NT100 "C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\um\credentialprovider.idl"`
`tlbImp2.exe credentialprovider.tlb /out:CredentialProvider.Interop.dll /unsafe /verbose /preservesig`

When I remove /verbose it will create the dll but the size is **17KB** rather than yours that is **48KB**, but also when added as a reference to my Custom Credential Provider project there are a ton of missing references which leads me to believe that the built CredentialProvider.Interop.dll is incomplete.

**Comment by phaetto** [comment]
Hey @DarrylIdle!

You will need to copy paste the dependencies in one file in order for this to work. Unfortunately I don't remember the details.

If you can wait for some months, I will have an extension of this open source project with all the C# code for the interop.
